File: server-sdks/sockudo-http-dotnet/SockudoServer/Event.cs
using System.Collections.Generic;

namespace SockudoServer
{
    /// <summary>
    /// V2 extras for publish events.
    /// </summary>
    public class MessageExtras
    {
        /// <summary>
        /// Flat key-value headers.
        /// </summary>
        public Dictionary<string, object> Headers { get; set; }

        /// <summary>
        /// Whether the event is ephemeral (not stored in history).
        /// </summary>
        public bool? Ephemeral { get; set; }

        /// <summary>
        /// An idempotency key for deduplication within extras.
        /// </summary>
        public string IdempotencyKey { get; set; }

        /// <summary>
        /// Whether the event should be echoed back to the sender.
        /// </summary>
        public bool? Echo { get; set; }
    }

    /// <summary>
    /// Represents an event for batch submission
    /// </summary>
    public class Event
    {
        /// <summary>
        /// The event name
        /// </summary>
        public string EventName { get; set; }

        /// <summary>
        /// The channel to which the event should be sent
        /// </summary>
        public string Channel { get; set; }

        /// <summary>
        /// An optional socket ID which should not receive the event
        /// </summary>
        public string SocketId { get; set; }

        /// <summary>
        /// The event data
        /// </summary>
        public object Data { get; set; }

        /// <summary>
        /// An optional idempotency key for deduplicating this event.
        /// When provided, it is included in the JSON body and sent as an X-Idempotency-Key header.
        /// </summary>
        public string IdempotencyKey { get; set; }

        /// <summary>
        /// Optional V2 extras for the event.
        /// </summary>
        public MessageExtras Extras { get; set; }
    }
}

File: server-sdks/sockudo-http-java/src/main/java/io/sockudo/rest/data/Event.java
package io.sockudo.rest.data;

import io.sockudo.rest.util.Prerequisites;

/**
 * POJO for JSON encoding of trigger batch events.
 */
public class Event {

    private final String channel;
    private final String name;
    private final Object data;
    private final String socketId;
    private final String idempotencyKey;
    private final MessageExtras extras;

    public Event(final String channel, final String eventName, final Object data) {
        this(channel, eventName, data, null, null, null);
    }

    public Event(final String channel, final String eventName, final Object data, final String socketId) {
        this(channel, eventName, data, socketId, null, null);
    }

    public Event(final String channel, final String eventName, final Object data, final String socketId, final String idempotencyKey) {
        this(channel, eventName, data, socketId, idempotencyKey, null);
    }

    public Event(final String channel, final String eventName, final Object data, final String socketId, final String idempotencyKey, final MessageExtras extras) {
        Prerequisites.nonNull("channel", channel);
        Prerequisites.nonNull("eventName", eventName);
        Prerequisites.nonNull("data", data);

        this.channel = channel;
        this.name = eventName;
        this.data = data;
        this.socketId = socketId;
        this.idempotencyKey = idempotencyKey;
        this.extras = extras;
    }

    public String getChannel() {
        return channel;
    }

    public String getName() {
        return name;
    }

    public Object getData() {
        return data;
    }

    public String getSocketId() {
        return socketId;
    }

    public String getIdempotencyKey() {
        return idempotencyKey;
    }

    public MessageExtras getExtras() {
        return extras;
    }
}
